BootloaderCorePkg/Stage2/Stage2Support.c

```diff
@@ -80,12 +80,20 @@ BoardNotifyPhase (
     PlatformService = (PLATFORM_SERVICE *) GetServiceBySignature (PLATFORM_SERVICE_SIGNATURE);
     // Add perf data to FPDT - SBL Performance Table
     Rsdp = (EFI_ACPI_5_0_ROOT_SYSTEM_DESCRIPTION_POINTER *)(UINTN)PcdGet32 (PcdAcpiTablesRsdp);
-    Fpdt = FindAcpiTableBySignature((EFI_ACPI_DESCRIPTION_HEADER *)(UINTN)Rsdp->RsdtAddress, SIGNATURE_32('F', 'P', 'D', 'T'), NULL);
-    if (Fpdt != NULL) {
-      DEBUG ((DEBUG_INFO, "FPDT found @ 0x%X, updating..\n", Fpdt));
-      PlatformService->AcpiTableUpdate((UINT8 *)Fpdt, Fpdt->Length);
+    if (Rsdp != NULL) {
+      Fpdt = FindAcpiTableBySignature((EFI_ACPI_DESCRIPTION_HEADER *)(UINTN)Rsdp->RsdtAddress, SIGNATURE_32('F', 'P', 'D', 'T'), NULL);
+      if (Fpdt != NULL) {
+        DEBUG ((DEBUG_INFO, "FPDT found @ 0x%X, updating..\n", Fpdt));
+        Status = PlatformService->AcpiTableUpdate((UINT8 *)Fpdt, Fpdt->Length);
```

Comment:
I think line 93 ~ 97 could be removed. If we want to know the FPDT update status, we could move line 86 after line 87 and print Status. e.g.:  DEBUG ((DEBUG_INFO, "Update FPDT @ 0x%X, Status =%r\n", Fpdt, Status));

Reply:
Thanks for the feedback!